### ISSUE TITLE
Toil hangs on service initiation

### ISSUE THREAD

**Issue by almussel** [BD2KGenomics/toil#548]
When I run the attached script with the command:
```
python bug_script.py --logDebug ./foo
```
it non-deterministically hangs, sometimes after starting the first service, sometimes after starting the second service. Sometimes it runs successfully to completion.

```
#!/usr/bin/env python2.7
# Audrey Musselman-Brown

"""
Debugging Script
"""

import sys
import argparse
from toil.job import Job


log = open("bug_python.log", 'a')


def initialize(job):
    log.write("initialize\n")
    log.flush()
    job.addChildJobFn(job1)


def job1(job):
    log.write("job1\n")
    log.flush()
    job.addService(Service1())
    job.addChildJobFn(job2)


def job2(job):
    log.write("job2\n")
    log.flush()
    job.addService(Service2())
    job.addChildJobFn(hello_world)


def hello_world(job):
    log.write("Hello World\n")


# SERVICE CLASSES

class Service1(Job.Service):

    def start(self):
        log.write("start Service1\n")
        log.flush()
        return
        
    def stop(self):
        log.write("stop Service1\n")
        log.flush()
        return

                
class Service2(Job.Service):

    def start(self):
        log.write("start Service2\n")
        log.flush()
        return

    def stop(self):
        log.write("stop Service2\n")
        log.flush()
        return


def main(args):
    
    parser = argparse.ArgumentParser()    
    Job.Runner.addToilOptions(parser)
    options = parser.parse_args()
    Job.Runner.startToil(Job.wrapJobFn(initialize), options)

if __name__=="__main__":
    sys.exit(main(sys.argv[1:]))
```

**Comment by almussel** [comment]
Issue is reproducible with SingleMachine on 3.1.0, but the script runs correctly on Mesos with 3.1.0.

**Comment by benedictpaten** [comment]
Is there a test PR for this? I'm ready to pick it up - but quick please.

**Comment by hannes-ucsc** [comment]
Since its SingleMachine only I should probably take this.

**Comment by hannes-ucsc** [comment]
If a job doesn't specify any core requirements, the default core requirement will be applied. The default core requirement can be configured via command line option `--defaultCores` but defaults to 1. On a two-core machine, two such jobs can run in parallel but the above DAG has three concurrent jobs: two services and the hello_word job at the end.

@almussel can you try to confirm this by reducing the requirement to .1 and running your script again on SingleMachine?

**Comment by almussel** [comment]
I just tried this:

```
python bug_script.py --defaultCores=0.1 ./foo
```

and got the following error:

```
No handlers could be found for logger "toil.resource"
INFO:toil.lib.bioio:Logging set at level: INFO
Traceback (most recent call last):
  File "bug_script.py", line 76, in <module>
    sys.exit(main(sys.argv[1:]))
  File "bug_script.py", line 73, in main
    Job.Runner.startToil(Job.wrapJobFn(initialize), options)
  File "/usr/local/lib/python2.7/dist-packages/toil/job.py", line 421, in startToil
    with setupToil(options, userScript=job.getUserScript()) as (config, batchSystem, jobStore):
  File "/usr/lib/python2.7/contextlib.py", line 17, in __enter__
    return self.gen.next()
  File "/usr/local/lib/python2.7/dist-packages/toil/common.py", line 418, in setupToil
    config.setOptions(options)
  File "/usr/local/lib/python2.7/dist-packages/toil/common.py", line 141, in setOptions
    setOption("defaultCores", h2b, iC(1))
  File "/usr/local/lib/python2.7/dist-packages/toil/common.py", line 99, in setOption
    % (varName, x))
RuntimeError: The defaultCores option has an invalid value: 0
```

**Comment by hannes-ucsc** [comment]
The above backtrace is indicative of another, unrelated bug which I'll just fix in the PR #588.

**Comment by hannes-ucsc** [comment]
Audrey and I confirmed that reducing the requirements to .1 prevents the hang, rendering the primary issue invalid.
